<?xml version="1.0"?>
<launch>

  <!-- Stowed arm joint positions -->
  <arg name="stowed_shou_yaw" default="0" />
  <arg name="stowed_shou_pitch" default="0" />
  <arg name="stowed_prox_pitch" default="0" />
  <arg name="stowed_dist_pitch" default="0" />
  <arg name="stowed_hand_yaw" default="0" />
  <arg name="stowed_scoop_yaw" default="0" />
  <arg name="stowed_grinder_yaw" default="0.0" />

  <group ns="planning">
    <arg name="debug" default="false" />
    <arg name="use_gui" default="false" />

    <include file="$(find ow_lander)/launch/planning_context.launch">
      <arg name="load_robot_description" value="true"/>
    </include>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="use_gui" value="$(arg use_gui)"/>
      <param name="zeros/j_shou_yaw" value="$(arg stowed_shou_yaw)"/>
      <param name="zeros/j_shou_pitch" value="$(arg stowed_shou_pitch)"/>
      <param name="zeros/j_prox_pitch" value="$(arg stowed_prox_pitch)"/>
      <param name="zeros/j_dist_pitch" value="$(arg stowed_dist_pitch)"/>
      <param name="zeros/j_hand_yaw" value="$(arg stowed_hand_yaw)"/>
      <param name="zeros/j_scoop_yaw" value="$(arg stowed_scoop_yaw)"/>
      <param name="zeros/j_grinder" value="$(arg stowed_grinder_yaw)"/>

      <rosparam param="source_list">[move_group/fake_controller_joint_states]</rosparam>
    </node>

    <!-- == movegroup, used to control the arm ========== -->
    <include file="$(find ow_lander)/launch/move_group.launch">
      <arg name="allow_trajectory_execution" value="true"/>
      <arg name="fake_execution" value="true"/>
      <arg name="info" value="true"/>
      <arg name="debug" value="$(arg debug)"/>
    </include>

    <!-- == launch arm specific rviz ============== -->
    <include file="$(find ow_lander)/launch/arm_sim_rviz.launch">
      <arg name="config" value="true"/>
      <arg name="debug" value="$(arg debug)"/>
    </include>

    <!-- == launch the path planner node ============== -->
    <node pkg="ow_lander" name="path_planning_commander" type="path_planning_commander.py" output="screen">
    </node>

    <!-- == launch the trajectory publishing node ========= -->
    <node pkg="ow_lander" name="trajectory_publisher" type="trajectory_publisher.py" output="screen">
    </node>

  </group>

</launch>
